.github/workflows/generate-test-duration.yml

```diff
@@ -0,0 +1,220 @@
+        if: inputs.os=='ubuntu-latest'
+        run: sudo apt install graphviz
+      - name: Install kubectl on Linux
+        run: |
+          curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+          sudo install -o root -g 0 -m 0755 kubectl /usr/local/bin/kubectl
+        if: inputs.os=='ubuntu-latest'
+      - name: Install K3D
+        run: |
+          curl -s https://raw.githubusercontent.com/rancher/k3d/main/install.sh | bash
+        if: runner.os!='Windows' && contains(inputs.test_environment, 'kubeflow')
+      - name: Login to Amazon ECR
+        id: login-ecr
+        run: |
+          aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin 715803424590.dkr.ecr.us-east-1.amazonaws.com
+        if: contains(inputs.test_environment, 'aws')
+      - name: Login to Amazon EKS
+        id: login-eks
+        run: |
+          aws eks --region us-east-1 update-kubeconfig --name zenml-ci-cluster --alias zenml-ci-aws-us-east-1
+        if: contains(inputs.test_environment, 'aws')
+      - name: Login to Google ECR
+        run: |
+          gcloud auth configure-docker --project zenml-ci
+        if: contains(inputs.test_environment, 'gcp')
+      - name: Login to Google GKE
+        uses: google-github-actions/get-gke-credentials@v2
+        with:
+          cluster_name: zenml-ci-cluster
+          location: us-east4
+          project_id: zenml-ci
+        if: contains(inputs.test_environment, 'gcp')
+      - name: Setup tmate session before tests
+        if: ${{ inputs.enable_tmate == 'before-tests' }}
+        uses: mxschmitt/action-tmate@v3.17
+        timeout-minutes: ${{ inputs.tmate_timeout }}
+      - name: Generate test duration file
+        continue-on-error: true
+            # Ubuntu integration tests run as 6 shards
+        run: |
+          bash scripts/test-coverage-xml.sh "" default "" "" store-durations
+      - name: Check difference in .test_duration
+        run: |
+          git diff --quiet || echo "Changes found"
+          if [ -n "$(git status --porcelain)" ]; then
+              # Generate feature branch name with the current date
+              export FEATURE_BRANCH="misc/update-test-duration-file-$(date +'%Y%m%d')"
+              git checkout -b $FEATURE_BRANCH
+              # Make your changes here
+              git add .
+              git commit -m "new changes on test duration file detected at $(date +'%Y%m%d')"
+              git push origin $FEATURE_BRANCH
+          else
+              echo "No changes in .test_duration"
+          fi
+      - name: Create PR if changes found
+        if: success()
+        run: |
+          if [ -n "$FEATURE_BRANCH" ]; then
+              gh pr create --base develop --head $FEATURE_BRANCH --title "Update `.test_duration` due to detected changes" --body "This PR was automatically created by the CI pipeline. Please review and merge if you agree."
```

Comment:
Please add at least one of us on the team as a reviewer in this command. Also add the 'tests' and 'internal' tags using the command. (There are flags available for this).